dune/gdt/local/numerical-fluxes/lax-friedrichs.hh

```diff
@@ -54,23 +60,44 @@ public:
 
   using BaseType::apply;
 
-  StateRangeType apply(const StateRangeType& u,
-                       const StateRangeType& v,
-                       const PhysicalDomainType& n,
-                       const XT::Common::Parameter& param = {}) const override final
+  StateType apply(const I& intersection,
+                  const LocalIntersectionCoords& x,
+                  const StateType& u,
+                  const StateType& v,
+                  const PhysicalDomainType& n,
+                  const XT::Common::Parameter& param = {}) const override final
   {
-    const auto lambda =
-        1. / std::max(this->flux().jacobian(u, param).infinity_norm(), this->flux().jacobian(v, param).infinity_norm());
-    return 0.5 * ((this->flux().evaluate(u, param) + this->flux().evaluate(v, param)) * n) + 0.5 * ((u - v) / lambda);
+    const auto local_flux = this->flux().local_function();
+    local_flux->bind(intersection.inside());
+    this->compute_entity_coords(intersection, x);
+    const auto lambda = 1.
+                        / std::max(local_flux->jacobian(x_in_inside_coords_, u, param).infinity_norm(),
+                                   local_flux->jacobian(x_in_outside_coords_, v, param).infinity_norm());
+    return 0.5
+               * ((local_flux->evaluate(x_in_inside_coords_, u, param)
+                   + local_flux->evaluate(x_in_outside_coords_, v, param))
```

Comment:
As an afterthought: to make the code a bit more readable, you could do something like:
```c++
// prepare
auto local_flux_inside = this->flux().local_function();
local_flux_inside->bind(intersection.inside());
auto local_flux_outside = this->flux().local_function();
local_flux_outside->bind(intersection.neighbor() ? intersection.outside() : intersection.inside());
// evaluate
const auto df_u = local_flux_inside->jacobian(x_in_inside_coords_, u, param);
const auto df_v = local_flux_outside->jacobian(x_in_outside_coords_, v, param);
const auto f_u = local_flux_inside->evaluate(x_in_inside_coords_, u, param);
const auto f_v = local_flux_outside->evaluate(x_in_outside_coords_, v, param);
// compute numerical flux
const auto lambda = 1. / std::max(df_u.infinity_norm(), df_v.infinity_norm());
return 0.5 * ((f_u + f_v) * n) + 0.5 * ((u - v) / lambda);
```
but just as a suggestion...

Reply:
Done.